src/compiler/checker.ts

```diff
@@ -35120,7 +35120,9 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             if (declaration.type) {
                 const typeNode = getEffectiveTypeAnnotationNode(declaration);
                 if (typeNode) {
-                    inferTypes(inferenceContext.inferences, getTypeFromTypeNode(typeNode), getTypeAtPosition(context, i));
+                    const source = addOptionality(getTypeFromTypeNode(typeNode), /*isProperty*/ false, hasQuestionToken(declaration));

```

Comment:
`hasQuestionToken(declaration)` isn't going to do the right thing for jsdoc optional parameters - instead, use the catch-all `isOptionalDeclaration`.

Reply:
👍 pushed out the appropriate change